include/pybind11/pytypes.h

```diff
@@ -1637,7 +1639,33 @@ class capsule : public object {
         }
     }
 
-    const char *name() const { return PyCapsule_GetName(m_ptr); }
+    const char *name() const {
+        const char *name = PyCapsule_GetName(m_ptr);
+        if ((name == nullptr) && PyErr_Occurred()) {
+            throw error_already_set();
+        }
+        return name;
+    }
+
+    /// Replaces a capsule's name *without* calling the destructor on the existing one.
+    void set_name(const char *new_name) {
+        if (PyCapsule_SetName(m_ptr, new_name) != 0) {
+            throw error_already_set();
+        }
+    }
+
+private:
+    static const char *get_name_no_throw(PyObject *o) {
```

Comment:
get_name_preserve_error
get_name_in_error_scope
get_name_with_error_scope
...
?

The "no throw" makes sense in the context of the development history of this PR, but not so much without that context.

Reply:
Yes, I like `get_name_in_error_scope`.